checkov/common/graph/graph_builder/local_graph.py

```diff
@@ -20,3 +19,63 @@ def __init__(self) -> None:
     @abstractmethod
     def build_graph(self, render_variables: bool) -> None:
         pass
+
+    def get_vertices_with_degrees_conditions(
+        self, out_degree_cond: Callable[[int], bool], in_degree_cond: Callable[[int], bool]
+    ) -> List[int]:
+        vertices_with_out_degree = {
+            vertex_index for vertex_index in self.out_edges.keys() if out_degree_cond(len(self.out_edges[vertex_index]))
+        }
```

Comment:
```suggestion
        vertices_with_out_degree = {
            vertex_index for vertex_index, vertex_value in self.out_edges.items() if out_degree_cond(len(vertex_value))
        }
```
same works for the `in_edges`